airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -93,21 +122,42 @@ def __init__(
         cluster_name: str,
         cluster_role_arn: str,
         resources_vpc_config: Dict,
+        compute: Optional[str] = DEFAULT_COMPUTE_TYPE,
         nodegroup_name: Optional[str] = None,
         nodegroup_role_arn: Optional[str] = None,
-        compute: Optional[str] = DEFAULT_COMPUTE_TYPE,
+        fargate_profile_name: Optional[str] = None,
+        fargate_pod_execution_role_arn: Optional[str] = None,
+        fargate_selectors: Optional[List] = None,
         aws_conn_id: str = DEFAULT_CONN_ID,
         region: Optional[str] = None,
         **kwargs,
     ) -> None:
+        if compute:
+            try:
+                if (compute == 'nodegroup') and not nodegroup_role_arn:
+                    raise ValueError(
+                        MISSING_ARN_MSG.format(compute=NODEGROUP_FULL_NAME, requirement='nodegroup_role_arn')
+                    )
+                if (compute == 'fargate') and not fargate_pod_execution_role_arn:
+                    raise ValueError(
+                        MISSING_ARN_MSG.format(
+                            compute=FARGATE_FULL_NAME, requirement='fargate_pod_execution_role_arn'
+                        )
+                    )
+            except KeyError:
```

Comment:
When this situation can happen?

Reply:
This needs to get changed to an `else`.  In a previous iteration there was an assignment in there that could fail and I missed this on the refactor.   

TODO:  Drop the `try` and make the `except` an `else`.

Reply:
Should be addressed in https://github.com/apache/airflow/pull/18645/commits/b577f159075d1c6b75d90c54303041b78025f122